src/App/ComplexGeoData.h

```diff
@@ -280,6 +280,19 @@ class AppExport ComplexGeoData: public Base::Persistence, public Base::Handled
     virtual void flushElementMap() const;
     //@}
 
+    /** @name Save/restore */
+    //@{
+    void Save (Base::Writer &writer) const;
+    void Restore(Base::XMLReader &reader);
+    void SaveDocFile(Base::Writer &writer) const;
+    void RestoreDocFile(Base::Reader &reader);
+    unsigned int getMemSize (void) const;
+    void setPersistenceFileName(const char *name) const;
+    virtual void beforeSave() const;
+    bool isRestoreFailed() const { return _restoreFailed; }
+    void resetRestoreFailure() const { _restoreFailed = true; }

```

Comment:
should not be const?

Reply:
I think it gets called from `const` methods, so has to be. `_restoreFailed` is mutable.